### PR TITLE
Updates to web-nodes to ignore Eclipse users and fix default JSHint found warnings

### PR DESCRIPTION
**Pull request by drwoods**
Similar to changes made in 606 for node-red project.
Ignore Eclipse created files.
Resolve some Eclipse JSHint warnings for curly brackets, missing commas, dead code in tests (used it.skip), added JSHint ignore for inner functions and fixed not using new for objects.

### CODE REVIEW

File: tfl/tfl-bus.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ module.exports = function(RED) {
         node.radius = n.radius;
 
         if(!node.stopCode1 || !node.lineID || node.stopCode1 === "unset" || node.lineID === "unset") {
-            node.error("No bus stops or bus line numbers have been configured.",msg);
```
Comment (drwoods) [inline tfl/tfl-bus.js]: `msg` was undefined here

File: test/strava/strava_spec.js [hunk 2]
```diff
@@ -104,7 +108,7 @@ describe('Strava node', function() {
         });
         
         if (nock) { // featues requiring HTTP communication/mocking
-            
+        	/*jshint -W082 */
```
Comment (drwoods) [inline test/strava/strava_spec.js]: ignore functions defined inside blocks

File: test/instagram/instagram_spec.js [hunk 3]
```diff
@@ -427,7 +443,7 @@ describe('instagram nodes', function() {
                 });
                 
             });
-            
+            /*jshint -W082 */
```
Comment (drwoods) [inline test/instagram/instagram_spec.js]: ignore functions defined in blocks

File: test/instagram/instagram_spec.js [hunk 4]
```diff
@@ -115,7 +119,7 @@ describe('instagram nodes', function() {
         });
         
         if (nock) { // featues requiring HTTP communication/mocking // TODO check if all tests require nock here
-            
+        	/*jshint -W082 */
```
Comment (drwoods) [inline test/instagram/instagram_spec.js]: ignore functions defined inside blocks

File: test/google/calendar_spec.js [hunk 5]
```diff
@@ -29,7 +29,7 @@ function TimeOffset(offsetSeconds) {
 }
 
 function ISOTimeString(offsetSeconds) {
-    return TimeOffset(offsetSeconds).toISOString();
+    return new TimeOffset(offsetSeconds).toISOString();
```
Comment (drwoods) [inline test/google/calendar_spec.js]: missing new of object types

File: test/delicious/delicious_spec.js [hunk 6]
```diff
@@ -116,7 +116,7 @@ describe('delicious nodes', function() {
                           function() {
                               var scope = nock('https://api.delicious.com:443')
                                   .get('/v1/posts/add?url=foobar&description=test&auth_token=undefined&shared=no&tags=testtag&extended=testdesc')
-                                    .reply(200, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\<code=\"done\"/>");
+                                    .reply(200, "<?xml version=\"1.0\" encoding=\"UTF-8\"?><code=\"done\"/>");
```
Comment (drwoods) [inline test/delicious/delicious_spec.js]: invalid HTML code

File: test/delicious/delicious_spec.js [hunk 7]
```diff
@@ -145,7 +145,7 @@ describe('delicious nodes', function() {
                           function() {
                               var scope = nock('https://api.delicious.com:443')
                                   .get('/v1/posts/add?url=foobar&description=test&auth_token=undefined&shared=no&tags=testtag&extended=testdesc')
-                                    .reply(401, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\<code=\"access denied\"/>");
+                                    .reply(401, "<?xml version=\"1.0\" encoding=\"UTF-8\"?><code=\"access denied\"/>");
```
Comment (drwoods) [inline test/delicious/delicious_spec.js]: invalid HTML code

File: test/box/box_spec.js [hunk 8]
```diff
@@ -227,9 +235,10 @@ describe('box nodes', function() {
                     });
                 });
         });
-        return; // TODO: finish below
```
Comment (drwoods) [inline test/box/box_spec.js]: unreachable code

File: test/box/box_spec.js [hunk 9]
```diff
@@ -227,9 +235,10 @@ describe('box nodes', function() {
...
+        
+        // TODO: finish below tests
 
-        it('should report no event when filepattern does not match', function(done) {
+        it.skip('should report no event when filepattern does not match', function(done) {
```
Comment (drwoods) [inline test/box/box_spec.js]: skip tests instead of using `return`

File: .gitignore [hunk 10]
```diff
@@ -1,3 +1,9 @@
-node_modules
 .npm
+.jshintignore
+.jshintrc
+.project
```
Comment (drwoods) [inline .gitignore]: ignore Eclipse generated project files for Node nature

File: .npmignore [hunk 11]
```diff
@@ -0,0 +1,7 @@
+.settings
```
Comment (drwoods) [inline .npmignore]: don't include any Eclipse files in the generated Node pkg

**Comment by drwoods** [comment]
Added some code comments for the not so obvious changes

**Comment by knolleary** [comment]
:+1:

**Comment by dceejay** [comment]
Why are we adding lots of extraneous ; to closing }  ? ... Mostly in first column, and not consistently either.

Causing a lot of unnecessary diff changes.

**Comment by drwoods** [comment]
@dceejay the default jshint settings for the Eclipse Node nature catches missing semicolons, as that can cause ambiguity and runtime errors.  I'll look at adding that and the curly bracket check to the jshint settings in the grunt build.

**Comment by knolleary** [comment]
@drwoods please don't rush to change the jshint settings we have in place. 

We choose not to enforce semicolons as we don't believe they are necessary - see http://jshint.com/docs/options/#asi for why.

**Comment by dceejay** [comment]
Hopefully you can set Eclipse to match the defaults we already use - rather than the other way round... 
As long as we have curly brace true then I think we are happy enough without semi-colons. 
The grunt build already has the required settings - so can you copy those into Eclipse ? Thanks

**Comment by knolleary** [comment]
The easier way to do it will be for us to extract our settings from
Gruntfile.js to a .jshintrc file that I think the eclipse tooling will pick
up.

**Comment by dceejay** [comment]
Well that is indeed what I have - but I also tend to run strict:true and unused:true locally which is a bit excessive in the general case so I didn't  - and also why it's in .gitignore ??

**Comment by knolleary** [comment]
It got added to .gitignore because it was one of the files the eclipse tooling generates and we didn't want it to get picked up - which was fine as we didn't have a .jshintrc of our own.

But, I'm now suggesting we move our standard set of rules out of the Gruntfile, into a .jshintrc file, remove it from .gitignore and be done with it.

If you want to run with a local customised version that's fine... but lets get the standard project one sorted.

**Pull request merged**